core/state/cache.go

```diff
@@ -0,0 +1,145 @@
+package state
+
+import (
+	"fmt"
+
+	"github.com/NethermindEth/juno/core/felt"
+)
+
+const (
+	// DefaultMaxLayers is the default maximum number of layers to keep in the cache
+	DefaultMaxLayers = 128
+)
+
+type diffCache struct {
+	storageDiffs      map[felt.Felt]map[felt.Felt]*felt.Felt // addr -> {key -> value, ...}
+	nonces            map[felt.Felt]*felt.Felt               // addr -> nonce
+	deployedContracts map[felt.Felt]*felt.Felt               // addr -> class hash
+}
+
+type stateCache struct {
+	diffs      map[felt.Felt]*diffCache // state root -> state diff
+	links      map[felt.Felt]felt.Felt  // child -> parent
+	oldestRoot felt.Felt                // root of the oldest layer in the cache
```

Comment:
It also means you don't have to keep this value since it will be the first element of the list